### PR TITLE
MBS-11862: Do not show deprecated relationship types with 0 uses

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11862

We've historically just removed link types once we no longer wanted them used and they had 0 uses, but that is a bad idea. It means the historical edits automatically show less info, since there's no type to load anymore, and it's just an unwanted loss of historical data.
As such, this allows us to keep them in the database, but not show them except when there's something to gain from doing so: for type editing / documenting, and for the edit search.

### CODE REVIEW

File: lib/MusicBrainz/Server/Data/LinkType.pm [hunk 1]
```diff
@@ -114,14 +114,27 @@ sub load
...
 sub get_tree
 {
-    my ($self, $type0, $type1) = @_;
+    my ($self, $type0, $type1, $get_deprecated_and_empty) = @_;
 
     my %id_to_obj;
     my @objs;
+    my $extra_condition = '';
+
+    unless ($get_deprecated_and_empty) {
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Data/LinkType.pm:123]: This might make things a little clearer:

```suggestion
    my ($self, $type0, $type1, %opts) = @_;

    my %id_to_obj;
    my @objs;
    my $extra_condition = '';

    unless ($opts{get_deprecated_and_empty}) {
```
then instead of doing `get_tree($type0, $type1, 1)` you would do `get_tree($type0, $type1, get_deprecated_and_empty => 1)`, which is more self-documenting.

File: lib/MusicBrainz/Server/Data/LinkType.pm [hunk 2]
```diff
@@ -114,14 +114,27 @@ sub load
...
+        $extra_condition = <<~'EOSQL';
+            AND NOT (
+                is_deprecated = TRUE
+                AND NOT EXISTS (
+                    SELECT 1 FROM link WHERE link.link_type = id
+                )
+            )
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Data/LinkType.pm:130]: The double negative is kinda hard to parse. How about:
```suggestion
            AND is_deprecated = FALSE
            AND EXISTS (SELECT 1 FROM link WHERE link.link_type = id)
```

Comment (reosarevok) [reply lib/MusicBrainz/Server/Data/LinkType.pm:130]: That's not what we want, though, it should show deprecated things as long as 1 exists in link, but I guess it could be simpler with those two conditions with an OR. Will do.

File: lib/MusicBrainz/Server/Data/LinkType.pm (hunk 1 above)
Comment (reosarevok) [reply lib/MusicBrainz/Server/Data/LinkType.pm:123]: Thanks, I wanted that but always forget how it goes :D
